core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala

```diff
@@ -48,7 +48,17 @@ private[ui] class ExecutorThreadDumpPage(parent: ExecutorsTab) extends WebUIPage
     val time = System.currentTimeMillis()
     val maybeThreadDump = sc.get.getExecutorThreadDump(executorId)
 
-    val content = maybeThreadDump.map { threadDump =>
+    val grepExp = Option(request.getParameter("grepexp"))
+
+    val filteredContent = maybeThreadDump.map { threadDump =>
+      threadDump.filter(thread =>
+        if (!grepExp.isDefined) {
+          true
+        } else {
+          thread.stackTrace.filter(_ >= ' ').matches(grepExp.get)
```

Comment:
The filter seems strange to me.  Won't stripping out newlines etc. lead to some weird matches?

Also, I think `String.matches` may not be what we want -- that matches against the entire string, not against any contained substring.  It would most likely mean that everyone would have to make their pattern `".*<pattern>.*"`.  I think we'd rather use `pattern.find()`, since that is more like "grep".  Should probably also be in multiline mode, so:

```scala
val grepExp = Option(request.getParamter("grepexp")).map(Pattern.compile(_, Pattern.MULTILINE))
...
grepExp.map { grep => grep.matcher(thread.stackTrace).find()}.getOrElse(true)
```